template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/knox/KnoxServiceConfigProvider.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.knox;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateComponentConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+@Component
+public class KnoxServiceConfigProvider implements CmTemplateComponentConfigProvider {
+
+    private static final String SAFETY_VALVE = "conf/gateway-site.xml_service_safety_valve";
+
+    private static final String DEFAULT_PROXY_NAME = "cdp-proxy";
```

Comment:
This can be fetched from the gateway object. We can avoid defining it at multiple places.

Reply:
nvm